### PR TITLE
fix(kit): correct default active month based on `min/max`

### LINKED ISSUE #3090: 🐞 - InputDate opens current month and not the month of the maximum date
### Which @taiga-ui/* package(s) are the source of the bug?

kit

### Please provide a link to a minimal reproduction of the bug

https://taiga-ui.dev/components/input-date/API?tuiMode=null&max$=1

### Is this issue blocking you?

Non-Blocking

### Description

If setting the maximum date for tui-input-date < the current month, then when you open the calendar, the current month of the current year is still opened, where all the days are locked (as they should be). Clicking on the arrow to go to the previous month immediately opens the month of the set maximum date, where you can choose something. So, in fact, one can still use the input, but the UX doesn’t seem to be great here.
It seems like it would be much nicer if the calendar would open the maximum possible month instead of the current one (when max month < current).

An example for this issue:
https://taiga-ui.dev/components/input-date/API?tuiMode=null&min$=1

image

### Angular version

11.2.13

### Taiga UI version

2.57.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3090

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/demo/src/modules/components/input-date/input-date.component.ts [hunk 1]
```diff
@@ -65,7 +65,12 @@ export class ExampleTuiInputDateComponent extends AbstractExampleTuiControl {
         ),
     };
 
-    minVariants = [TUI_FIRST_DAY, new TuiDay(2017, 2, 5), new TuiDay(1900, 0, 1)];
+    minVariants = [
+        TUI_FIRST_DAY,
+        new TuiDay(2017, 2, 5),
+        new TuiDay(1900, 0, 1),
+        TuiDay.normalizeOf(new Date().getFullYear() + 3, 1, 1),
```
Comment (waterplea) [inline projects/demo/src/modules/components/input-date/input-date.component.ts:72]: nit: why not just `new`?

**Review by waterplea — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/demo-integrations/cypress/tests/desktop/kit/input-date-range/input-date-range.cy.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-describe(`InputDateRange`, () => {
+describe(`InputDateTime`, () => {
```
Comment (waterplea) [inline projects/demo-integrations/cypress/tests/desktop/kit/input-date-range/input-date-range.cy.ts:1]: file is called `input-date-range` 🤔

**Review by waterplea — APPROVED** [review]

File: projects/demo-integrations/cypress/tests/desktop/kit/input-date-range/input-date-range.cy.ts (hunk 2 above)
Comment (splincode) [reply projects/demo-integrations/cypress/tests/desktop/kit/input-date-range/input-date-range.cy.ts]: fixed

File: projects/cdk/utils/calendar/compute-active-year-or-month.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import {TuiYear} from '@taiga-ui/cdk/date-time';
+
+export function tuiComputeActiveYearOrMonth<T extends TuiYear>(
```
Comment (waterplea) [inline projects/cdk/utils/calendar/compute-active-year-or-month.ts:3]: Let's call it `tuiDateClamp`.

Comment (splincode) [reply projects/cdk/utils/calendar/compute-active-year-or-month.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**